src/client/visualizations/scatterplot/utils/get-plotting-data.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017-2022 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as d3 from "d3";
+import memoizeOne from "memoize-one";
+import { Dataset, Datum } from "plywood";
+import { Essence } from "../../../../common/models/essence/essence";
+import { ConcreteSeries } from "../../../../common/models/series/concrete-series";
+import { Stage } from "../../../../common/models/stage/stage";
+import { selectFirstSplitDatums } from "../../../utils/dataset/selectors/selectors";
+import { LinearScale, pickTicks } from "../../../utils/linear-scale/linear-scale";
+
+const MARGIN = 40;
+const X_AXIS_HEIGHT = 50;
+const Y_AXIS_WIDTH = 50;
+const BREAKPOINT_SMALL = 768;
+const TICK_COUNT_SMALL = 4;
+const TICK_COUNT_DEFAULT = 10;
+
+interface PlottingData {
+  xSeries: ConcreteSeries;
+  ySeries: ConcreteSeries;
+  xScale: LinearScale;
+  yScale: LinearScale;
+  xTicks: number[];
+  yTicks: number[];
+  plottingStage: Stage;
+  scatterplotData: Datum[];
+}
+
+function getTicksCountByDimension(size: number): number {
+  if (size < BREAKPOINT_SMALL) {
```

Comment:
I'm always wondering if such names give us anything? I won't argue with Uncle Bob right now, but I still wonder ...

Reply:
Understandable names FTW 😄  I guess it is a matter of convention, I prefer to give names to numbers in general. Some are obvious, like screen size here, and perhaps don't need it. Other are not much so and giving them a proper name makes the code easier to read.